File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxLevelSum(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();

        queue.add(root);
        
        int lavel = 1 ;
        int maxLavel = 1;
        int maxSum = Integer.MIN_VALUE;

        while(!queue.isEmpty()) {
            int n = queue.size();
            int sum = 0;

            TreeNode node = new TreeNode();
            for(int i = 0 ; i<n ; i++) {
                node = queue.poll();
                sum = sum + node.val;

                if(node.left!=null) {
                    queue.add(node.left);
                }

                if(node.right!=null) {
                    queue.add(node.right);
                }

            }

            if(sum>maxSum) {
                maxSum = sum;
                maxLavel = lavel;
            }

            lavel++;

            
        }

        return maxLavel;
    }
}

File: 1116-maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.py
class Solution(object):
    def maxLevelSum(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        if not root:
            return 0

        from collections import deque

        queue = deque([root])
        level = 1
        max_level = 1
        max_sum = float('-inf')

        while queue:
            n = len(queue)
            current_sum = 0

            for _ in range(n):
                node = queue.popleft()
                current_sum += node.val

                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

            if current_sum > max_sum:
                max_sum = current_sum
                max_level = level

            level += 1

        return max_level

